src/Listener.cpp

```diff
@@ -0,0 +1,146 @@
+/*
+ * File: Listener.cpp
+ * Author: Craig Cogdill
+ * Created: August 14, 2015 1:08PM
+ */
+
+#include <memory>
+#include <g2log.hpp>
+#include <thread>
+#include <sstream>
+#include "Listener.h"
+#include <Alien.h>
+#include <Rifle.h>
+
+namespace {
+
+/**
+* RAII delete the string after it has been sent through the Rifle/Vampire queue back to the 
+* Notifier. This can't be a member function of Listener.
+*/
+void ZeroCopyDelete(void*, void* data) {
+  std::string* theString = reinterpret_cast<std::string*>(data);
+  delete theString;  
+} 
+} // namespace
+      
+Listener::Listener(const std::string& notificationQueue, const std::string& handshakeQueue, const std::string& program)
+   : mNotificationQueueName(notificationQueue),
+     mHandshakeQueueName(handshakeQueue),
+     mProgramName(program) {}
+
+/*
+ * Initialize a new Alien to point to the
+ *    given ipc file. This will enable a
+ *    Listener instance to receive
+ *    messages from the associated Notifier Shotgun
+ * 
+ * @return a bool if successful
+ */
+bool Listener::Initialize() {
+   if (mQueueReader.get() == nullptr) {
+      mHandshakeQueue = std::move(CreateFeedbackShooter());
+      if (!mHandshakeQueue) {
+         LOG(WARNING) << "Could not initialize Listener Handshake queue.";
+         return false;
+      }
+
+      mQueueReader.reset(new Alien);
+      try {
+         mQueueReader->PrepareToBeShot(mNotificationQueueName);
+      } catch (std::exception& e) {
+         LOG(WARNING) << "Listener Caught exception: " << e.what();
+         Reset();
+      }
+   }
+   return (mQueueReader.get() != nullptr && mHandshakeQueue.get() != nullptr);
+}
+
+/*
+* Create a Rifle to fire a confirmation message
+*     back to the original Notifier
+*
+* @return a feedback shooter of type Rifle*
+*/  
+std::unique_ptr<Rifle> Listener::CreateFeedbackShooter() {
+   LOG(INFO) << "Creating feedback handshake " << mHandshakeQueueName 
+             << " for thread #" << std::this_thread::get_id();
+   const size_t feedbackQSize = 100;
+   auto shooter = std::unique_ptr<Rifle>(new Rifle(mHandshakeQueueName));
+   shooter->SetOwnSocket(false);
+   shooter->SetHighWater(feedbackQSize);
+   if (!shooter->Aim()) {
+      LOG(WARNING) << "Failed to initialize feedback queue : " << mHandshakeQueueName;
+      shooter.reset();
+   }
+   return shooter;
+}
+
+/*
+ * Reset the member Alien to nullptr
+ */
+void Listener::Reset() {
+   mQueueReader.reset(nullptr);
```

Comment:
What happens to the  mHandshakeQueue = std::move(CreateFeedbackShooter()); that was created in Initialize() ? Is that cleaned up?